server/src/services/soccer.service.ts

```diff
@@ -116,4 +120,38 @@ export const soccerService = {
 
         return validatedData.response;
     },
+
+    transformFieldsToActualGenerationParams: async (
+        fieldsToTransform: Pick<PackagesGenerationParamsFromFreeText, 'teams' | 'league'>
+    ) => {
+        const { league, teams } = fieldsToTransform;
+        let actualLeague: League | undefined;
+        let actualTeams: Team[] | undefined;
+
+        if (league) {
+            const leagues = await soccerService.getLeaguesByName(league);
+            if (leagues.length > 0) {
+                actualLeague = leagues[0].league;
+            }
+        }
+
+        if (teams) {
+            actualTeams = await Promise.all(
+                teams.map(async (team) => {
+                    const teamData = await soccerService.getTeamsWithVenueByName(team.name);
```

Comment:
```suggestion
                    const teamData = await soccerService.getTeamsByName(team.name);
```

use that `getTeamsByName` and then you wont even need to make the `.map`